src/Components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,277 @@
+  );
+
+  class CustomToolbar extends Toolbar {
+    handleNavigation = (value: string) => {
+      switch (value) {
+        case "Previous":
+          this.navigate("PREV");
+          break;
+        case "Today":
+          this.navigate("TODAY");
+          break;
+        case "Next":
+          this.navigate("NEXT");
+          break;
+      }
+    };
+
+    render() {
+      return (
+        <div className="rbc-toolbar">
+          <div className={styles.toolbarNav}>
+            <Toggle
+              name="NavButtons"
+              values={["Previous", "Today", "Next"]}
+              defaultValue=""
+              setChecked={this.handleNavigation}
+            />
+          </div>
+          <p className="rbc-toolbar-label">{this.props.label}</p>
+          <Toggle
+            name="viewSelect"
+            values={["Month", "Week", "Day"]}
+            setChecked={changeView}
+            defaultValue={toggleValue}
+          />
+        </div>
+      );
+    }
+
+    navigate = (action) => {
+      this.props.onNavigate(action);
+    };
+  }
+
+  const onView = useCallback(
+    (newView) => {
+      setView(newView);
+      setToggleValue("Day");
+    },
+    [setView]
+  );
+
+  //Week Label
+  const dayRangeHeaderFormat = (
+    { start, end },
+    culture,
+    localizer: momentLocalizer
+  ) => {
+    const sameMonth = localizer.eq(start, end, "month");
+    const sameYear = localizer.eq(start, end, "year");
+    const formatter = `DD ${sameMonth ? "" : "MMM"} ${sameYear ? "" : "YYYY"}`;
+
+    return (
+      localizer.format(start, formatter, culture) +
+      " – " +
+      localizer.format(
+        end,
+        localizer.eq(start, end, "month") ? "DD MMMM YYYY" : "DD MMM YYYY",
+        culture
+      )
+    );
+  };
+
+  //Day Label
+  const dayHeaderFormat = (date: Date, culture, localizer: momentLocalizer) =>
+    localizer.format(date, "dddd DD MMMM, YYYY", culture);
+
+  const changeView = (value: string) => {
+    if (value === "Month") {
+      setView(Views.MONTH);
+      setToggleValue("Month");
+    }
+    if (value === "Week") {
+      setView(Views.WEEK);
+      setToggleValue("Week");
+    }
+    if (value === "Day") {
+      setView(Views.DAY);
+      setToggleValue("Day");
+    }
+  };
+
+  const togglePopup = () => {
+    setShow(!show);
+  };
+
+  const eventClick = (event: any) => {
+    const newEvent = {
+      id: event.id,
+      title: event.title,
+      start: convertToDateTimeLocalString(event.start),
+      end: convertToDateTimeLocalString(event.end),
+    } as PopupEvent;
+    setEventPopup(newEvent);
+    togglePopup();
+  };
+
+  const handleSelectSlot = (event: any) => {
+    let endTime = event.end;
+    if (event.action === "click") {
+      endTime = moment(event.start).add(defaultEventLength, "m").toDate();
+    }
+    const newEvent = {
+      id: myEvents.slice(-1)[0].id + 1, //get next ID available after last element. Should be safe?
```

Comment:
Better to use a prototype here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast

The method you're currently using doesn't handle an empty list